pexec/managed_process.go

```diff
@@ -102,6 +107,13 @@ func (p *managedProcess) ID() string {
 	return p.id
 }
 
+func (p *managedProcess) Status() error {
+	p.mu.Lock()
+  defer p.mu.Unlock()
+  err := p.cmd.Process.Signal(syscall.Signal(0))
```

Comment:
What if we wrote a `status` method in each of `managed_process_unix.go` and `managed_process_windows.go` where the former does `p.cmd.Process.Signal(syscall.Signal(0))` and the latter just `return errors.New("status not available on windows, signals not supported")`?

Reply:
Actually, I think I misread the golang docs about this. As this is just a signal and not an Interrupt, this should be compatible with windows as well. 

https://pkg.go.dev/os#Process.Signal

Reply:
So we should be okay keeping it in the main managed process file, without the warning

Reply:
Nice; SGTM!